src/pages/workspaces/workspace/analysis/utils/disk-utils.ts

```diff
@@ -91,6 +92,12 @@ export const getCurrentPersistentDisk = (runtimes: Runtime[], persistentDisks: P
     _.last(_.sortBy('auditInfo.createdDate', _.filter(({ id, status }) => status !== 'Deleting' && !_.includes(id, attachedIds), persistentDisks)))
 }
 
+export const getReadyPersistentDisk = (persistentDisks: PersistentDisk[]): PersistentDisk | undefined => {
+  // returns persistent disk if one exists and is in ready status
+  const disk = persistentDisks ? persistentDisks[0] : undefined
```

Comment:
Is always grabbing the head of the list, then checking that for readiness, a back-compatibility move? I assume so & that's why we can't do `return persistentDisks.find(disk => disk.status === diskStatuses.ready.leoLabel)`

Reply:
I think that should work!

My original fix just stopped us from using getCurrentPersistentDisk (I recommend looking at that function just for a bit of context) @mlilynolting

Reply:
yeah as Nathan said, I was building off of his original functionality, but yeah I can incorporate that logic, seems more readable!

Reply:
> Actually... seems like there's an issue trying to create a runtime too soon after deleting it... 🤔
> 
> This happened with deleting runtime only
> 
> I wonder if this needs to be addressed / can be addressed in this PR?

I think this is what [this ticket](https://broadworkbench.atlassian.net/browse/IA-4272) is trying to address, related but slightly different bug